packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -503,6 +594,107 @@ test.serial('writeGranulesFromMessage() saves the same values to DynamoDB, Postg
   t.deepEqual(omit(translatedPgRecord, omitList), omit(esRecord, omitList));
 });
 
+test.serial('writeGranulesFromMessage() given a partial granule updates only provided fields', async (t) => {
+  const {
+    collectionCumulusId,
+    cumulusMessage,
+    esGranulesClient,
+    granuleModel,
+    granulePgModel,
+    knex,
+    executionCumulusId,
+    providerCumulusId,
+    granuleId,
+  } = t.context;
+
+  await writeGranulesFromMessage({
+    cumulusMessage,
+    executionCumulusId,
+    providerCumulusId,
+    knex,
+    granuleModel,
+  });
+
+  t.true(await granuleModel.exists({ granuleId }));
+  t.true(await granulePgModel.exists(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  ));
+  t.true(await esGranulesClient.exists(granuleId));
+
+  const originalpgGranule = await granulePgModel.get(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  );
+
+  const { updatedPgGranuleFields, pgGranule } = await updateGranule(t, true);
+
+  // FUTURE:
+  // 1. 'created_at' is updated during PUT/PATCH
+  // 2. 'published' defaults to false if not provided in the payload
+  // 3. 'duration' comes from the workflow and will be reset on update
+  // 4. 'product_volume' comes from a files object on the payload, which may not exist
+  //   in the case of partial granule updates
+  const omitList = [
+    'cumulus_id',
+    'updated_at',
+    'created_at',
+    'published',
+    'timestamp',
+    'duration',
+    'product_volume',
+  ];
+
+  // Postgres granule matches expected updatedGranule
```

Comment:
Shouldn't we check all of the datastores to ensure consistency across all datastores?

Reply:
Yes. I split it up into two tests, one here for "Did we update the granule correctly" and another for "Do all of the datastores still match?"

See this assertion:
https://github.com/nasa/cumulus/pull/3082/files/9794f966250a13dee5704a469f3aa48c65487646#diff-e4b95ee0e3b92433a8e670895c1e8da32d083c3ca64e31761d4bb353cbb791a7R1615-R1626